bitforge/privkey.py

```diff
@@ -1,126 +1,178 @@
-import random, struct, binascii, collections
-import network, utils
-from errors import *
-from network import Network
-from pubkey import PublicKey
-from address import Address
-from encoding import *
+import collections
 
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives.asymmetric import ec
 
-rng     = random.SystemRandom()
-KEY_MAX = utils.generator_secp256k1.order()
+from bitforge import encoding, error, network, pubkey, tools
 
-def random_secret():
-    return rng.randint(1, KEY_MAX - 1)
 
+class Error(error.BitforgeError):
+    pass
 
-def find_network(value, attr = 'name'):
-    try:
-        return Network.get_by_field(attr, value)
-    except:
-        raise PrivateKey.UnknownNetwork(attr, value)
 
+class InvalidExponent(Error):
+    """Invalid private key secret exponent"""
+
+
+class InvalidEncoding(Error):
+    pass
+
+
+BasePrivateKey = collections.namedtuple('PrivateKey', [
+    'key',  # Elliptic curve private key
+    'network',  # Bitcoin-compatible network
+    'compressed',  # Whether the public key should be serialized in compressed format
+])
 
-BasePrivateKey = collections.namedtuple('PrivateKey',
-    ['secret', 'network', 'compressed']
-)
 
 class PrivateKey(BasePrivateKey):
+    """Bitcoin private key."""
+
+    def __new__(cls, key, network=network.default, compressed=True):
+        """Create a Bitcoin private key from an EC private key."""
+
+        return super(PrivateKey, cls).__new__(cls, key, network, compressed)
+
+    @classmethod
+    def generate(cls, network=network.default, compressed=True, backend=default_backend()):
```

Comment:
pep8:
> Don't use spaces around the = sign when used to indicate a keyword argument or a default parameter value.

`backend` is the cryptography library backend, i.e. the backing library for crypto utils, like OpenSSL, CommonCrypto, etc. More info: https://cryptography.io/en/latest/hazmat/backends/